NARFO_BE/NARFO_BE/Controllers/MembersController.cs

```diff
@@ -1,126 +1,183 @@
+            }else
             {
-                return NotFound();
+                user = await _context.Member.FirstOrDefaultAsync(member => member.Username == model.Username && Encryption.VerifyPassword(model.Password,member.Password));
             }
-
-            return members;
-        }
-
-        // PUT: api/Members/5
-        [HttpPut("{id}")]
-        public async Task<IActionResult> PutMembers(int id, Members members)
+                if(user == null)
+                {
+                    return BadRequest(new { status = "Failed", message = "Invalid Login"});
+                } else
+                {
+                    var tokenString = BuildToken(user);
+                    return Ok(new { status = "Success", token = tokenString });
+                }
+     }
+     
+        [HttpGet("all/user")]
+        public  async Task<ActionResult<MemberPrototype>> GetMembersEmail()
         {
-            if (id != members.Id)
+            List<MemberPrototype> endpoint = new List<MemberPrototype>();
+            foreach (Member member in  await _context.Member.ToArrayAsync())
             {
-                return BadRequest();
+               if(member.Username !=  null && member.Email != null )
+               endpoint.Add(new MemberPrototype(member.Username,member.Email));       
             }
-
-            _context.Entry(members).State = EntityState.Modified;
-
-            try
+            if(endpoint == null)
+            {
+                return BadRequest(new { status = "failed", error = "Failed to connect" });
+            }
+          return      Ok(new { status = "success", members=endpoint });
+        }
+       
+        [HttpGet("all")]
+        public async Task<ActionResult<IEnumerable<Member>>> GetMembers() {
+            return await _context.Member.ToListAsync();
+        }
+     
+        [HttpGet("{id}")]
+        public async Task<ActionResult<Member>> GetMembers(string id){
+            var Members = await _context.Member.FindAsync(id);
+            if (Members == null)
             {
-                await _context.SaveChangesAsync();
+                return BadRequest(new { status = "failed", error = "Failed to connect" });
             }
-            catch (DbUpdateConcurrencyException)
+            return Ok(new { status = "success", members = Members });
+        }
+ 
+        [HttpGet("get/{id}")]
+        public async Task<ActionResult<Member>> GetMembers(int id)
+        {
+            var amembers = await _context.Member.FindAsync(id); //gets the member with matching id
+            if (amembers == null) { return BadRequest(new { status = "failed", error = "Failed to connect" }); }//fail response 
+            return Ok(new { status = "success", members = amembers });//success response
+        }
+
+        [HttpPost("verify/username")]
+        public async Task<ActionResult<Member>> VerifyUsername([FromBody]Member Member)
```

Comment:
this should probably be done in one api. with a model with both username and email

Reply:
merge the VerifyUsername and VerifyUser in to one api